scenarios/excluded/dependency-excludes-non-contiguous-range-of-compatible-versions.toml

```diff
@@ -0,0 +1,42 @@
+name = "dependency-excludes-non-contiguous-range-of-compatible-versions"
+description = "There is a non-contiguous range of compatible versions for the requested package `a`, but another dependency `c` excludes the range. This is the same as `dependency-excludes-range-of-compatible-versions` but some of the versions of `a` are incompatible for another reason e.g. dependency on non-existent package `d`."
+
+[root]
+requires = ["a", "b>=2.0.0,<3.0.0", "c"]
+
+[expected]
+satisfiable = false
+explanation = "Only the `2.x` versions of `a` are available since `a==1.0.0` and `a==3.0.0` require incompatible versions of `b`, but all available versions of `c` exclude that range of `a` so resolution fails."
+
+[packages.a.versions."1.0.0"]
```

Comment:
Can we use nesting for package versions? Is that easier to read?

Reply:
The pre-existing `fork/*.toml` seem to also use this syntax, we can attempt to change it across though.

e.g. `fork/filter-sibling-dependencies.toml`
```
[packages.a.versions."4.3.0"]
[packages.a.versions."4.4.0"]
[packages.b.versions."1.0.0"]
requires = ["d==1.0.0"]
[packages.c.versions."1.0.0"]
requires = ["d==2.0.0"]
[packages.d.versions."1.0.0"]
[packages.d.versions."2.0.0"]
```